api/controllers/shabads.js

```diff
@@ -438,64 +437,91 @@ const getShabad = (ShabadIDQ, sinceDate = null) =>
-              UNION
-            (SELECT 'next' as navigation,ShabadID FROM Shabad WHERE VerseID= ? LIMIT 1);`;
-              conn
-                .query(q1, [rows[0].ID - 1, rows[rows.length - 1].ID + 1])
-                .then(rows1 => {
-                  let previous = null;
-                  let next = null;
-                  rows1.forEach(row => {
-                    if (row.navigation === 'previous') {
-                      previous = row.ShabadID;
-                    }
-                    if (row.navigation === 'next') {
-                      next = row.ShabadID;
-                    }
-                  });
-                  const navigation = {
-                    previous,
-                    next,
-                  };
-
-                  resolve({
-                    shabadInfo,
-                    count: verses.length,
-                    navigation,
-                    verses,
-                  });
-                  conn.end();
-                })
-                .catch(err => reject(err));
+              let curShabadID = -1;
+              let counter = 0;
+              rows.forEach(row => {
+                if (row.ShabadID !== curShabadID) {
+                  curShabadID = row.ShabadID;
+                  output.shabadIds.push(row.ShabadID);
+                  counter = output.shabads.push([]) - 1;
+                }
+                output.shabads[counter].push(row);
+              });
+
+              const outputShabadPromises = output.shabads.map(getShabadSingle);
+              output.shabads = await Promise.all(outputShabadPromises);
+              resolve(output);
             } else {
               resolve({});
             }
+            if (conn) conn.end();
           })
           .catch(err => reject(err));
       })
       .catch(err => reject(err));
   });
+
+const getShabadSingle = async rows => {
+  const shabadInfo = lib.getShabadInfo(rows[0]);
+  const verses = rows.map(lib.prepVerse);
+  const navigation = await getShabadNavigation(rows[0].ID - 1, rows[rows.length - 1].ID + 1);
+
+  return {
+    shabadInfo,
+    count: verses.length,
+    navigation,
+    verses,
+  };
+};
+
+const getShabadNavigation = async (prevLine, nextLine) => {
+  let conn;
+  try {
+    conn = await pool.getConnection();
+    const q1 = `(SELECT 'previous' as navigation,ShabadID FROM Shabad WHERE VerseID = ? LIMIT 1)
+                UNION
+                (SELECT 'next' as navigation,ShabadID FROM Shabad WHERE VerseID= ? LIMIT 1);`;
+    const rows1 = await conn.query(q1, [prevLine, nextLine]);
+    let previous = null;
+    let next = null;
+    rows1.forEach(row => {
+      if (row.navigation === 'previous') {
+        previous = row.ShabadID;
+      }
+      if (row.navigation === 'next') {
+        next = row.ShabadID;
+      }
+    });
+    return {
+      previous,
+      next,
+    };
+  } catch (err) {
+    error(err);
+  } finally {
+    if (conn) conn.end();
```

Comment:
nit - prefer `if` with curly brackets

if you don't want to change here, I will update later and standardize eslint rules across repos.

Reply:
yeah I was copying whats done elsewhere in the same file and repo